### PR TITLE
GFORMS-2396 - Update error messages for "overseasAddress" and "postco…

### PR DESCRIPTION
**Pull request by DursunGundogan**
…deLookup" to use address by default

### CODE REVIEW

File: app/uk/gov/hmrc/gform/addresslookup/AddressLookupController.scala [hunk 1]
```diff
@@ -629,8 +629,8 @@ class AddressLookupController(
     fastForward: List[FastForward]
   ): Action[AnyContent] =
     auth.authAndRetrieveForm[SectionSelectorType.Normal](formTemplateId, maybeAccessCode, OperationWithForm.EditForm) {
-      implicit request => implicit l => cache => implicit sse => _ =>
-        mkSyntheticFormModelOptics(formComponentId, cache, maybeAccessCode) {
+      implicit request => implicit l => cache => implicit sse => realFormModelOptics =>
+        mkSyntheticFormModelOptics(formComponentId, cache, maybeAccessCode, realFormModelOptics) {
```
Comment (lxol) [inline app/uk/gov/hmrc/gform/addresslookup/AddressLookupController.scala:633]: Is that for the case when we enter the address manually - to make "error" properties the same as parent component properites?

Comment (DursunGundogan) [reply app/uk/gov/hmrc/gform/addresslookup/AddressLookupController.scala:633]: Yes, when a user wants to enter an address manually, it creates a synthetic form component in this function. I think we need to copy errorShortName/errorShortNameStart from the `postcodelookup` component.

File: app/uk/gov/hmrc/gform/addresslookup/AddressLookupController.scala [hunk 2]
```diff
@@ -704,7 +707,9 @@ class AddressLookupController(
       editable = true,
       submissible = true,
       derived = false,
-      errorMessage = None
+      errorMessage = None,
```
Comment (lxol) [inline app/uk/gov/hmrc/gform/addresslookup/AddressLookupController.scala:710]: should we add errorMessage from the parent component as well?

Comment (DursunGundogan) [reply app/uk/gov/hmrc/gform/addresslookup/AddressLookupController.scala:710]: There is no AC about `errorMessage`. I'm not sure we need to add it

Comment (lxol) [reply app/uk/gov/hmrc/gform/addresslookup/AddressLookupController.scala:710]: @alexhurst don't you think it makes sense to mention in the ticket that if errorMessage is configured - the "manual address" page should respect that.
Also as a side note I guess the "manual address" page is not part of the error reporting. We should probably have a ticket to fix that.

Comment (lxol) [reply app/uk/gov/hmrc/gform/addresslookup/AddressLookupController.scala:710]: @alexhurst Actually I think there is a bigger problem here. 
```      "fields": [
        {
          "type": "postcodeLookup",
          "id": "postcode1",
          "label": "Address",
          "shortName": "Business address",
          "chooseAddressLabel": "Choose your business address",
          "confirmAddressLabel": "Confirm your business address"
        }
```
The error properties here will be applies to `postcode` page. The ticket applies them to "manual address" lookup page which is not in the form but synthetically generated and there is no way to to specify them explicitly.

Comment (alexhurst) [reply app/uk/gov/hmrc/gform/addresslookup/AddressLookupController.scala:710]: @lxol I think that is the problem with defining an errorMessage. So for this ticket defining the errorShortNames and only using them on the manual address entry page should work. I would not worry about errorMessage for now and then we can do another ticket if it is required. Or we introduce an errorMessage but that only applied to the manual address entry page.

**Review by lxol — APPROVED** [review]

**Pull request merged**